Access Engineering.js

```diff
@@ -0,0 +1,258 @@
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, url) {
+	if(url.includes('book') && url.match('chapter')) {
+		return 'bookSection';
+	} else if(url.includes('book')) {
+		return 'book';
+	} else if(url.includes('case-study') ||
+			  url.includes('video') ||
+			  url.includes('calculator') ||
+			  url.includes('tutorial')) {
+		return 'webpage';
+	}
+	return false;
+}
+
+function doWeb(doc, url) {
+	scrape(doc, url);
+}
+
+function scrape(doc, url) {
+	let item;
+	let main = doc.querySelector('.page-layout--mhe-layout');
+	
+	switch(detectWeb(doc, url)) {
+	  /* Meta Data: Title, Author, Edition, Publisher,
+				  Date, ISBN, URL, Abstract */
+	  case 'book':
+		  item = new Zotero.Item('book');
+		  
+		  // Title
+		  let book__title = text(main, '.h2').split(', ');
+		  item.title = book__title[0];
+		  
+		  // Edition
+		  let bookEdition = (item.title).split(', ');
+		  item.edition = bookEdition[1];
+
+		  // Publisher
+		  item.publisher = text(main, '.field--name-copyright-holder p');
+
+		  // Date
+		  item.date = text(main, '.field--name-copyright-year p');
+
+		  // ISBN
+		  item.ISBN = text(main, '.field--name-isbn-ebook-compact p');
+			  
+		  break;
+	  
+	  
+	  /* Meta Data: Title, Author, Book Title, Edition, 
+	  Publisher, Date, ISBN, URL, Abstract */
+	  case 'bookSection':
+		  item = new Zotero.Item('bookSection');
+		  
+		  // Title
+		  item.title = text(main, '.row--chapter-tabs .content-title');
+		  
+         // Book Title
+         let book__title = text(main, '.h2').split(', ');
+		  item.bookTitle = book__title[0];  
+         
+         // Edition
+		  let bookEdition = (item.title).split(', ');
+		  item.edition = bookEdition[1];
+
+		  // Publisher
+		  item.publisher = text(main, '.field--name-copyright-holder p');
+
+		  // Date
+		  item.date = text(main, '.field--name-copyright-year p');
+
+		  // ISBN
+		  item.ISBN = text(main, '.field--name-isbn-ebook-compact p');
+		  break;
+          
+      // Item Type: Web Page (Video, Spreadsheet, Case Study and Tutorial)
+	  /* Meta Data: Title, Author, Date, URL, Abstract */
+	  case 'webpage':
+		  item = new Zotero.Item('webpage');
+		  
+		  // Title
+		  item.title = text(main, '.h2');
+		  
+		  // Date
+		  item.date = text(main, '.field--name-date-print-publication');
+		  break;          
+          
+	}
+	
+	// Author
+	for (let author of main.querySelectorAll('.contrib-group-authors a')) {
+		item.creators.push(ZU.cleanAuthor(author.textContent, 'author'));
+	}
+	
+	// URL
+	item.url = url;
+	
+	// Abstract
+	if(doc.querySelectorAll('.highwire-truncate-text__full_content').length > 0) {
+		item.abstractNote = text(main, '.highwire-truncate-text__full_content .short-view__content');
+	} else {
+		item.abstractNote = text(main, '.highwire-truncate-text__content');
+	}
+		
+	//Zotero.debug(item);
+	
+	item.attachments.push({
+		title: 'Snapshot',
```

Comment:
No need to add a snapshot for catalog entries.

Reply:
Removed snapshot.